litestar/_signature/models/attrs_signature_model.py

```diff
@@ -238,30 +242,52 @@ def __init__(self) -> None:
 _converter: Converter = Converter()
 
 
-def _extract_exceptions(e: Any) -> list[ErrorMessage]:
+def _get_message_from_traceback(exc: StructureException, connection: ASGIConnection) -> ErrorMessage | None:
+    err_format = [
+        line for line in traceback.format_exception(type(exc), value=exc, tb=exc.__traceback__) if key_re.search(line)
+    ]
+
+    # handles nested structures (e.g. a model attribute references another
+    # model) by going through all exceptions in the exception group
+    key_matches = []
+    for line in compact(err_format):
+        if k := key_re.findall(line)[0][0].strip():
+            key_matches.append(k)
+
+    exception = exc
+
+    if isinstance(exception, cattrs.ClassValidationError):
+        # ensure we get the actual cause of the exception by getting the last
+        # exception in the group
+        while hasattr(exception, "exceptions"):
+            exception = exception.exceptions[-1]
+
+    message = {"key": ".".join(key_matches), "message": str(exception)}
+
+    if key_matches and key_matches[0] != "data" and key_matches[0] in connection.query_params:
+        message["key"] = f"query_params.{message['key']}"
```

Comment:
Not sure how the team feels about this, but I prepended `query_params` (where applicable) in the spirit of adding clarity around the value that's got a problem.

Reply:
Hi @rseeley,

First off - thanks for contributing 😁

Secondly, I don't like this change. I think the parameter name is sufficient. 

Furthermore, you do not handle header and cookie parameters, no? 

Finally - whatever changes are done on the attrs side will have to also be done on the pydantic side. Our intention is to wrap error handling so it does not leak implementation details.

Reply:
I suggest we add another field to `ErrorMessage` for reporting this:

```py
class ErrorMessage(TypedDict):
    key: str
    message: str
    source: Literal["body", "query", "cookie", "header"]
```
Something like that.

Reply:
I like that idea. How do you feel on that, @Goldziher ?

Nice catch on the other data types (header, cookie). I'll update to include those.

Reply:
Sure